core/module_test.cc

```diff
@@ -0,0 +1,201 @@
+#include "module.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <gtest/gtest.h>
+
+#include "utils/cdlist.h"
+
+// Mocking out misc things  ------------------------------------------------
+
+__thread struct worker_context ctx = {};
+
+struct task *task_create(Module *m, void *arg) {
+  return nullptr;
+}
+
+void task_destroy(struct task *t) {}
+
+void task_attach(struct task *t, struct tc *c) {}
+void task_detach(struct task *t) {}
+
+struct snobj *snobj_str(const char *s) { return nullptr; }
+struct snobj *snobj_nil() { return nullptr; }
+
+struct snobj *snobj_err_details(int err, struct snobj *details, const char *fmt,
+                                ...) {
+  return nullptr;
+}
+
+void *mem_alloc(size_t size) {
+  void *ptr = malloc(size);
+
+  if (ptr) memset(ptr, 0, size);
+
+  return ptr;
+}
+
+void *mem_realloc(void *ptr, size_t size) { return realloc(ptr, size); }
+
+void mem_free(void *ptr) { free(ptr); }
+
+// -------------------------------------------------------------------------
+
+class AcmeModule : public Module {
+  public:
+    AcmeModule() : Module() {}
+
+    static const gate_idx_t kNumIGates = 1;
+    static const gate_idx_t kNumOGates = 1;
+
+    static const Commands<Module> cmds;
+
+    void Foo(struct snobj *arg) { n += 1; }
+
+    int n = {};
+};
+
+const Commands<Module> AcmeModule::cmds = {
+  {"foo", MODULE_FUNC &AcmeModule::Foo, 0}
+};
+
+int create_acme(const char *name, Module **m) {
+  EXPECT_EQ(1, ModuleBuilder::all_module_builders().size());
+  EXPECT_EQ(1, ModuleBuilder::all_module_builders().count("AcmeModule"));
```

Comment:
You might make this line and the one above ASSERT_EQ, because the rest of the test depends on them passing.